api/src/handlers.rs

```diff
@@ -24,9 +24,38 @@ struct PaginatedResponse {
     data: Vec<Value>,
 }
 
+pub fn check_table_exists(table: &str, tables: &[String]) -> Option<HttpResponse> {
+    if !tables.contains(&table.to_string()) {
+        Some(HttpResponse::NotFound().json(json!({
+            "error": format!("Table '{}' not found", table),
+            "valid_tables": tables,
+            "help": "Refer to the API documentation for valid table names."
+        })))
+    } else {
+        None
+    }
+}
+
 #[get("/tables")]
-pub async fn list_tables(data: web::Data<AppState>) -> impl Responder {
-    HttpResponse::Ok().json(&*data.tables)
+pub async fn list_tables(
+    query: web::Query<PaginationParams>,
+    data: web::Data<AppState>,
+) -> impl Responder {
+    let total_count = data.tables.len() as i64;
+    let pagination = Pagination::new(query, total_count);
+
+    let start = pagination.offset as usize;
+    let end = (start + pagination.limit as usize).min(data.tables.len());
+
+    let paginated_tables = &data.tables[start..end];
```

Comment:
i don't expect tables to grow so much as to require pagination, but this keeps responses consistent, so this is a good change.